### PR TITLE
[SDK] `buyProtectData()`

### PR DESCRIPTION
**Pull request by cedric25**

### CODE REVIEW

File: packages/sdk/src/lib/dataProtectorSharing/buyProtectedData.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    if (collection) {
+      tx = await sharingContract.buyProtectedDataForCollection(
+        protectedData.collection.id, // _collectionTokenIdFrom
+        protectedData.id,
+        collection.id, // _collectionTokenIdTo
+        vAppAddress || DEFAULT_PROTECTED_DATA_SHARING_APP,
+        {
+          // TODO: See how we can remove this
+          gasLimit: 900_000,
+        }
```
Comment (Le-Caignec) [inline packages/sdk/src/lib/dataProtectorSharing/buyProtectedData.ts:82]: you missed the value field : 
```suggestion
        {
           value: // price of the sale
          // TODO: See how we can remove this
          gasLimit: 900_000,
        }
```

File: packages/sdk/src/lib/dataProtectorSharing/buyProtectedData.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+      );
+    } else {
+      tx = await sharingContract.buyProtectedData(
+        protectedData.collection.id, // _collectionTokenIdFrom
+        protectedData.id,
+        userAddress,
+        {
+          // TODO: See how we can remove this
+          gasLimit: 900_000,
+        }
```
Comment (Le-Caignec) [inline packages/sdk/src/lib/dataProtectorSharing/buyProtectedData.ts:92]: idem

**Review by Le-Caignec — CHANGES_REQUESTED** [review]
Good Job, just some comments 👍🏻

File: packages/sdk/src/lib/dataProtectorSharing/buyProtectedData.ts (hunk 1 above)
Comment (cedric25) [reply packages/sdk/src/lib/dataProtectorSharing/buyProtectedData.ts]: Good catch indeed, thanks

**Review by Le-Caignec — APPROVED** [review]

**Pull request merged**